src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -104,18 +103,22 @@ trait RegularSync extends BlockBroadcast {
         val blockNumber = appStateStorage.getBestBlockNumber()
         val request = GetBlockHeaders(Left(blockNumber + 1), blockHeadersPerRequest, skip = 0, reverse = false)
         waitingForActor = Some(context.actorOf(
-          SyncBlockHeadersRequestHandler.props(peer, etcPeerManager, peerEventBus, request, resolveBranches = false)))
+          SyncBlockHeadersRequestHandler.props(peer, peerResponseTimeout, etcPeerManager, peerEventBus, request, resolveBranches = false)))
       case None =>
         log.debug("No peers to download from")
         scheduleResume()
     }
   }
 
   private def handleBlockBranchResolution(peer: Peer, message: Seq[BlockHeader]) =
-    //todo limit max branch depth? [EC-248]
     if (message.nonEmpty && message.last.hash == headersQueue.head.parentHash) {
       headersQueue = message ++ headersQueue
-      processBlockHeaders(peer, headersQueue)
+      if (headersQueue.length > branchResolutionMaxDepth) {
```

Comment:
So please correct me if I'm misreading the code here:
- start with 1 header which is missing the parent
- ask for 12 additional headers
- we have `13 < maxDepth` headers
- try to resolve and still parent not found
- ask for 12 more
- we have `25 > maxDepth`, we give up

And let's say the non-orphan header was the 20th in line. But we haven't checked it because the `maxDepth` condition short-circuited.